src/backends/gpu/multi_event_stream.hpp

```diff
@@ -58,18 +59,20 @@ class multi_event_stream_base {
         n_nonempty_stream_(1)
     {}
 
+    // The list of events must be sorted in ascending order of index,
+    // with events of equal index further time-sorted.
```

Comment:
Can't we just say 'sorted first by index and then by time', as used below? Or is this phrasing not common?